src/bucket/BucketSnapshot.cpp

```diff
@@ -314,8 +314,11 @@ LiveBucketSnapshot::scanForEviction(
                 // for persistent entries, we need to make sure we evict the
                 // newest version of the entry, since it will be persisted in
                 // the hot archive. So we add persistent entries to our DB
-                // lookup to find the newest version.
-                if (isPersistentEntry(le.data))
+                // lookup to find the newest version. Note that there was a
+                // bug in protocol 23 where this check was not performed.
+                if (isPersistentEntry(le.data) &&
```

Comment:
Can we also skip any scanning if eviction is disabled completely? (per [this](https://github.com/stellar/stellar-core-internal/issues/402#issuecomment-3429421778) comment)

Reply:
I don't think we should. It would add more complexity to the scan logic, and there are only about a week of ledgers where this optimization would be useful, so I don't think it's worth the extra code. There should be no reason to disable eviction in the future (other than bugs)